File: go/LeetCode/chalkReplacer.go
package main

func main() {
	
}

func chalkReplacer(chalk []int, k int) int {
	sum:=0
	for _,v := range chalk{
		sum+=v
	}
	mod:=k%sum
	for i:=0;i<len(chalk);i++ {
		if mod-chalk[i]<0 {
			return i
		}
		mod-=chalk[i]
	}
	return -1
}

File: python3/LeetCode/chalkReplacer.py
from typing import List


class Soultion:
    def chalkReplacer(self, chalk:List[int], k:int)->int:
        total = sum(chalk)
        mod = k%total
        for i in range(0, len(chalk)):
            if mod - chalk[i]<0:
                return i
            mod-=chalk[i]
        return -1